### PR TITLE
release-21.2: sql: allow auto commit when rows written limit is reached, not exceeded

### PR DESCRIPTION
**Pull request by yuzefovich**
Backport 1/1 commits from #70196.

/cc @cockroachdb/release

---

We recently added the support of the rows written/read guardrails.
Originally, those numbers meant that once reached, an error would be
emitted. In a0f68a17f935cbe1daf77a764f341adf01983ed2 we adjusted the
guardrails so that the numbers meant the largest allowed counts.
However, we forgot to update on place - namely, if "rows written"
erring guardrail is enabled, we have to check the count before allowing
the auto commit of all mutation operations. This meant that in case the
limit is reached but not exceeded, we would mistakenly lose the auto
commit optimization.

Release note: None

### CODE REVIEW

**Comment by yuzefovich** [comment]
Will wait for an explicit approval to merge this.

**Review by rytaft — APPROVED** [review]
:lgtm:

Since this isn't fixing a release blocker, I'd probably wait to merge this until 21.2.1

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/70203#-:-MjbFCiY5hGl3x9Doy3D:b4zpu76)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @yuzefovich)

**Comment by yuzefovich** [comment]
Oh, yeah, I got confused - I had no intention of merging this for 21.2.0. (I was thinking of #70179 when typed out my last comment.)

**Pull request merged**
